src/cascadia/TerminalApp/TerminalWarnings.h

```diff
@@ -0,0 +1,54 @@
+/*++
+Copyright (c) Microsoft Corporation
+Licensed under the MIT license.
+
+Module Name:
+- TerminalWarnings.h
+
+Abstract:
+- This file contains definitions for warnings, errors and exceptions used by the
+  Windows Terminal
+
+Author(s):
+- Mike Griese - August 2019
+
+--*/
+#pragma once
+
+namespace TerminalApp
+{
+    // SettingsLoadWarnings are scenarios where the settings contained
+    // information we knew was invalid, but we could recover from.
+    enum class SettingsLoadWarnings : uint32_t
+    {
+        MissingDefaultProfile = 0,
+        DuplicateProfile = 1
+    };
+
+    // SettingsLoadWarnings are scenarios where the settings had invalid state
+    // that we could not recover from.
+    enum class SettingsLoadErrors : uint32_t
+    {
+        NoProfiles = 0
+    };
+
+    // This is a helper class to wrap up a SettingsLoadErrors into a proper
+    // exception type.
+    class TerminalException : public std::runtime_error
+    {
+    public:
+        TerminalException(const SettingsLoadErrors& error) :
+            std::runtime_error{ nullptr },
+            _error{ error } {};
+
+        // We don't use the what() method - we want to be able to display
+        // localizable error messages. Catchers of this exception should use
+        // _GetErrorText to get the localized exception string.

```

Comment:
Where is this `_GetErrorText`? Also, wouldn't providing a `what()` at least help someone figure out what's going on in the debugger moreso than providing nothing even if it's not localized?

Reply:
I added some placeholder text. I f you really want, we could file a task to try and make that text specific to the exception value